src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_params.py

```diff
@@ -548,6 +548,8 @@ def register_common_storage_account_options(context):
 for item in ['file', 'blob']:
     register_cli_argument('storage {} url'.format(item), 'protocol', help='Protocol to use.', default='https', **enum_choice_list(['http', 'https']))
     register_source_uri_arguments('storage {} copy start'.format(item))
+    for command in ['download', 'download-batch', 'upload', 'upload-batch']:
+        register_cli_argument('storage {} {}'.format(item, command), 'progress_callback', ignore_type, validator=add_progress_callback)
```

Comment:
The essence of your change is using a parameter on the custom command and a validator to achieve a "dependency injection". This is the side effect of this approach. I do think we should look at knack to see if there is room to improve the overall mechanism of the core to allow dependency injection without explicit manipulate the arguments.

/cc @tjprescott